python/sglang/test/test_layernorm.py

```diff
@@ -109,5 +109,63 @@ def test_gemma_rms_norm(self):
                 self._run_gemma_rms_norm_test(*params)
 
 
+class TestLayerNorm(CustomTestCase):
+    DTYPES = [torch.half, torch.bfloat16, torch.float32]
+    PARAM_DTYPES = [torch.half, torch.bfloat16, torch.float32]
+    NUM_TOKENS = [7, 83, 1024]
+    HIDDEN_SIZES = [768, 769, 770, 771, 5120, 5124, 5125, 5126, 8192, 8199]
+    USE_AFFINE = [False, True]
+    USE_BIAS = [False, True]
+    SEEDS = [0]
+
+    @classmethod
+    def setUpClass(cls):
+        if not torch.cuda.is_available():
+            raise unittest.SkipTest("CUDA is not available")
+        torch.set_default_device("cuda")
+
+    def _run_layer_norm_test(self, num_tokens, hidden_size, use_affine, use_bias, dtype, seed, param_dtype):
+        torch.manual_seed(seed)
+
+        layer = LayerNorm(hidden_size, elementwise_affine=use_affine, bias=use_bias, dtype=param_dtype)
+        if use_affine:
+            layer.weight.data.normal_(mean=1.0, std=0.1)
+            if use_bias:
+                layer.bias.data.normal_(mean=0.0, std=0.1)
+        
+        scale = 1 / (2 * hidden_size)
+        x = torch.randn(num_tokens, hidden_size, dtype=dtype) * scale
+
+        with torch.inference_mode():
+            ref_out = layer.forward_native(x)
+            out = layer(x)
```

Comment:
In many of the cases, `layer(x)` would dispatch to `layer.forward_native(x)` so you're comparing the same code.